sympy/physics/quantum/state.py

```diff
@@ -665,7 +666,7 @@ def _eval_innerproduct(self, bra, **hints):
                 return S.Zero # i.e. Integer(0)
 
             if is_zero is None:
-                return None
+                return KroneckerDelta(arg, bra_arg)
```

Comment:
I'm not sure about this.

Reply:
The point of this is that $\langle i | j \rangle$ should be simplified to the Kronecker Delta for the case where they are orthonormal. In fact, maybe this entire method should be just a wrapper to the Kronecker Delta function. Leaving it as simply $\langle i | j \rangle$ is very annoying to perform summation over because you would have to replace it with $\delta_{ij}$ anyway to further simplify.